### PR TITLE
Deprecate Fisheries & Finfish models

### PR DESCRIPTION
**Pull request by davemfish**
Finfish Aquaculture & Fisheries models are being deprecated. This PR reflects all the items listed in our deprecations checklist wiki. The reporting module was a dependency of only Finfish, so that's also removed here.

The fisheries model was used in a number of `cli` tests, so many of the extra changes here are substituting in different models in those tests.

We'll leave #719 open after this is merged, as there are a couple last items on the checklist to do after the PR.

# Checklist
- [x] Updated HISTORY.rst (if these changes are user-facing)

- [x] Updated the user's guide (if needed)

Related PRs to be merged with this one:
https://bitbucket.org/natcap/invest-sample-data/pull-requests/21/task-deprecate-fish
https://bitbucket.org/natcap/invest-test-data/pull-requests/24/task-deprecate-fish
https://github.com/natcap/invest.users-guide/pull/52

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -50,6 +50,12 @@ Unreleased Changes (3.10)
       (i.e. lakes).
     * Added feature to accept raster (in addition to vector) habitat layers.
     * Changed one intermediate output (geomorphology) from SHP to GPKG.
+* Fisheries
```
Comment (emlys) [inline HISTORY.rst:53]:
```suggestion
* Fisheries and Fisheries HST
```

**Review by emlys — CHANGES_REQUESTED** [review]
Awesome @davemfish! Just had a tiny suggestion

**Comment by davemfish** [comment]
@emlys done!

**Review by emlys — APPROVED** [review]
Hooray! So satisfying to see so much code deleted lol

**Comment by phargogh** [comment]
Wow, I only just noticed just _how_ much code was removed here.  Let us pause for a melancholy moment in their memory.

**Pull request merged**